interfaces/prompting/requestrules/requestrules_test.go

```diff
@@ -0,0 +1,1553 @@
+package requestrules_test
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/interfaces/prompting"
+	"github.com/snapcore/snapd/interfaces/prompting/patterns"
+	"github.com/snapcore/snapd/interfaces/prompting/requestrules"
+	"github.com/snapcore/snapd/logger"
+	"github.com/snapcore/snapd/testutil"
+)
+
+func Test(t *testing.T) { TestingT(t) }
+
+type noticeInfo struct {
+	ruleID prompting.IDType
+	data   map[string]string
+}
+
+func (ni *noticeInfo) String() string {
+	return fmt.Sprintf("{\n\truleID: %s\n\tdata:   %#v\n}", ni.ruleID, ni.data)
+}
+
+type requestrulesSuite struct {
+	testutil.BaseTest
+
+	defaultNotifyRule func(userID uint32, ruleID prompting.IDType, data map[string]string) error
+	defaultUser       uint32
+	ruleNotices       []*noticeInfo
+}
+
+var _ = Suite(&requestrulesSuite{})
+
+func (s *requestrulesSuite) SetUpTest(c *C) {
+	s.defaultUser = 1000
+	s.defaultNotifyRule = func(userID uint32, ruleID prompting.IDType, data map[string]string) error {
+		c.Check(userID, Equals, s.defaultUser)
```

Comment:
This should be throwing errors for various `s.defaultUser + 1` users, but it doesn't, unless there's another error in the test. Not sure why this would be.

Reply:
Perhaps this is causing the intermittent failures on some unit test runs?
```
ok  	github.com/snapcore/snapd/interfaces/prompting/requestprompts	0.045s
OOPS: 34 passed, 14 FAILED
--- FAIL: Test (0.40s)
FAIL
FAIL	github.com/snapcore/snapd/interfaces/prompting/requestrules	0.408s
```